### PR TITLE
Feature/nvidia gpu os

### PR DESCRIPTION
**Pull request by radireddy**
### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

## Title of the Pull Requests 
Support NVIDIA GPU OS (NVIDIA GPU Cloud) while creating VM

### Description of the Change
Show NVIDIA GPU OS in preinstalled applications and distribution + apps, only for T1 instances
Add below validations

1. NVIDIA OS can only be used with T1 instances
2. Can not update the existing server (without T1 instance) with NVIDIA OS image

### Benefits
The customer can preinstall NVIDIA GPU OS in VM

### Possible Drawbacks
none

### Applicable Issues
MBP-154

### CODE REVIEW

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 1]
```diff
@@ -1436,6 +1467,21 @@ angular.module('managerApp')
         return _.find(CLOUD_INSTANCE_HAS_GUARANTEED_RESSOURCES, elem => elem === flavorType);
       };
 
+      /**
+       * We dont support OS and flavour change at same time.
+       * NVIDIA OS supported only in t1 and t2 inatances.
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]:
```suggestion
       * NVIDIA OS supported only in t1 and t2 instances.
```

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 2]
```diff
@@ -376,6 +376,37 @@ angular.module('managerApp')
...
+        let selectedFlavourInFilterList = false;
+        if (!_.isEmpty(restrictedFlavors)) {
+          self.displayData.categories = _.filter(
+            self.displayData.categories,
+            (category) => {
+              const currentCategory = category;
+              currentCategory.flavors = _.filter(
+                category.flavors,
+                (flavor) => {
+                  const isRestricted = _.indexOf(restrictedFlavors, flavor.shortType) > -1;
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]:
```suggestion
                  const isRestricted = _.includes(restrictedFlavors, flavor.shortType);
```

File: client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js [hunk 3]
```diff
@@ -376,6 +376,37 @@ angular.module('managerApp')
...
+                  if (!selectedFlavourInFilterList
+                    && isRestricted
+                    && selectedFlavour.id === flavor.id) {
+                    selectedFlavourInFilterList = true;
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]: Please only use `filter` to filter, you'd better use `find` to find items

Comment (radireddy) [reply client/app/cloud/project/compute/infrastructure/virtualMachine/addEdit/cloud-project-compute-infrastructure-virtualMachine-addEdit.controller.js]: I was not able to map your comment with code.  In this case, I am using _.filter to filter items from an array not to find an item. Are you sure your comment belongs to this code? can you please verify again.

**Comment by jleveugle** [comment]
Hello @radireddy,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Comment by radireddy** [comment]
@antleblanc, can you merge this PR? its in open state from a very long time

**Pull request merged**